### ISSUE TITLE
Create targetNamespace if it does not exist

### ISSUE THREAD

**Issue by zimme** [k3s-io/helm-controller#43]
Helm v2 defaults to creating the namespace if it doesn't exist when installing a chart.

Helm v3 on the other hand does not do this as it's seems to be praxis in the k8s eco system not to error out on missing namespace. However It seems as though helm v3 added a --create-namespace flag that could be used to create the namespace. It's available since 3.2.0 if I understood it correctly.

Would you accept a PR that uses that flag as the default?

**Comment by thecrazyrussian** [comment]
There is a limitation in the default create-namespace functionality I outlined in the rfe here: https://github.com/helm/helm/issues/8570
That makes it less useful for Rancher backed clusters. tl;dr; no annotations and labels support.

We could also add a wrapper on our side when a helm create-namespace is defined in the spec to optionally include annotations and labels and trigger the namespace creation before executing the helm release.

**Comment by fredleger** [comment]
i would suggest implementing in two steps:
- supports `--create-namespace = true`
- later support annotations

As i see it it can still be useful for us without labels or annotation.
The use case was to boostrap a cluster without any access yet. A pitty to see it failing because of missing ns (i can still add it to the manifests right, but why ?)

**Comment by jasine** [comment]
any progress?

**Comment by riuvshyn** [comment]
any updates on this?

**Comment by sergey-kudriavtsev** [comment]
Maybe something is already born?

**Comment by gioppoluca** [comment]
any update? this is needed!

**Comment by brandond** [comment]
You still haven't mentioned which chart you're referring to. I assume you're talking about the Rancher Monitoring charts? Those are meant to be deployed by Rancher itself, not directly via additionalManifest on the downstream cluster.

**Comment by gioppoluca** [comment]
Yes as I wrote we use rancher-monitoring-crd from https://charts.rancher.io.

The thing you say is most strange and utterly new since we discussed monitoring deployment with rancher support and this never came out.
Also this is from fleet examples:
https://github.com/rancher/fleet-examples/blob/master/single-cluster/helm-multi-chart/rancher-monitoring/fleet.yaml

What differs from an operation done by fleet and one done by the additionaManifest? A chart is a chart, the operation behind it is a helm operation.
I think that if charts from https://charts.rancher.io are not meant to be used through fleet it should be stated clearly.  But Rancher should also document how to automatically set a cluster with those charts since we need to find the UI integrations and are not happy of having to deploy the prometheus operator as it is.

**Comment by brandond** [comment]
Fleet and the Helm controller that is built in to K3s/RKE2 function very differently. Most of the stuff in Rancher expects that charts  in the catalog are going to be deployed by Fleet. If you're going to use Rancher, you should do things the Rancher way.
